### DISCUSSION TITLE
[Calendar] How do you set the "default" date?

### DISCUSSION THREAD

**Discussion by matthew-dean** [primefaces/community#1748] [PrimeVue]
When you have a calendar field and click to open the calendar, it defaults to the current date / time. The calendar picker we're using is for UTC time. How do you set the "current" default time for the drop-down calendar WITHOUT setting a value for the field?

**Comment by Terrijoo** [comment]
I'm having the same issue currently (e.g. having two fields, one "date from" and one "date until", having a default time of 23:59:59 would actually be quite convienent), did you (or anyone) ever found a solution?

**Reply by Terrijoo** [reply]
Actually found a way with PrimeVue and pt options, for anyone having the same task/issue.
Not sure, if this the perfect way, but it does seem to work in my case.

Template:
```
<DatePicker
    [...] other options
    :pt="computedPTOptions"
/>
```

Script:
```
const computedPTOptions = computed(() => ({
    root: (options) => {
        options.state.currentHour = 23;
        options.state.currentMinute = 59;
        options.state.currentSecond = 59;
        return;
    }
}));
```

**Comment by ISL-Tiemann** [comment]
I'm having the same issue. We wanted to default to 00:00 (when no value is set) since in our context the default value of the current time is a confusing for the user. 

`defaultValue` is no option, since it is written to the underlying InputText component and visible in the UI.

I did it the same way:

Script:
```ts
import type { PassThrough } from "@primevue/core";
import type { DatePickerPassThroughOptions } from "primevue/datepicker";

// PassThrough for the PrimeVue DatePicker Component, to set the hour and minute to 0 when the component has no value.
// The default is the minute and hour of the current time of the browser.
const overlayTimeToZeroWhenEmptyPt: PassThrough<DatePickerPassThroughOptions> = {
  root: ({ state, props }) => {
    if (!props.modelValue) {
      state.currentHour = 0
      state.currentMinute = 0
    }
  }
}
```

Template:
```vue
<DatePicker
  [...] other options
  :pt="overlayTimeToZeroWhenEmptyPt"
  />
```

Would be nice if there was a property to set the default date for the picker overlay only. This way, it feels a bit hacky.

**Comment by cadilhac** [comment]
I second the request. I have a picker for a start date and one for an end date of a period. When setting the start date and the end date is still null, I would like to set the default date in the end date's calendar to be the start date.
